Include/internal/pycore_lock.h

```diff
@@ -26,7 +26,7 @@ extern "C" {
 // 0b11: locked and has parked threads
 //
 // Typical initialization:
-//   PyMutex m = (PyMutex){0};
+//   PyMutex m = {0};
```

Comment:
I don't think we should make this change. The comment is for variables that need explicit initialization, such as local variables and fields in heap allocated structures. Static variables are implicitly initialized to zero in C.

Reply:
Thanks for the review! For local variables, should we remove the cast `(PyMutex)`? Seems it is needless in modern C.

Reply:
Ah, I've got the point!